### PR TITLE
Created ExternalPathProvider. When USE_DOWNLOADS_DIR_AS_HOME is true, downloads is the home dir.

### PR DESCRIPTION
**Pull request by teamagam5**
Created ExternalPathProvider.
Supplying paths depending on configurable constant: USE_DOWNLOADS_DIR_AS_HOME.

### CODE REVIEW

File: data/src/main/java/com/teamagam/gimelgimel/data/common/ExternalPathProvider.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    private final Context mContext;
+    private final File mAppHomeDir;
+
+    @Inject
+    public ExternalPathProvider(Context context) {
+        mContext = context;
+        mAppHomeDir = new File(getAppHomeDirPath());
+    }
+
+    public File getExternalFilesDir() {
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/common/ExternalPathProvider.java]: "Cache" returned path once and retrieve it on subsequent requests

File: data/src/main/java/com/teamagam/gimelgimel/data/common/ExternalPathProvider.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (Constants.USE_DOWNLOADS_DIR_AS_HOME) {
+            return getExternalDir(FILES_DIR_NAME);
+        } else {
+            return mContext.getExternalFilesDir(null);
+        }
+    }
+
+    public File getExternalCacheDir() {
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/common/ExternalPathProvider.java]: Same

File: data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java [hunk 3]
```diff
@@ -3,7 +3,8 @@
 import android.graphics.Bitmap;
 
 public class Constants {
-    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev1.herokuapp.com";
+    public static final String MESSAGING_SERVER_URL = "http://ggmessaging-dev2.herokuapp.com";
+    public static final boolean USE_DOWNLOADS_DIR_AS_HOME = true;
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java]: I'd place it not in between two related constants xD

File: data/src/main/java/com/teamagam/gimelgimel/data/images/ImageUtils.java [hunk 4]
```diff
@@ -20,17 +19,11 @@
...
-     * needed for temp file from camera.
-     *
-     * @param context
-     * @return
-     * @throws IOException
-     */
-    public static Uri getTempImageUri(Context context){
+    public static Uri getTempImageUri(ExternalPathProvider externalPathProvider){
 
         //temp dir
```
Comment (BarAgam) [inline data/src/main/java/com/teamagam/gimelgimel/data/images/ImageUtils.java]: useless comment !@# 😡

**Review by BarAgam — APPROVED** [review]
Groovy work.
Very shagadelic.

File: data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java (hunk 3 above)
Comment (teamagam5) [reply data/src/main/java/com/teamagam/gimelgimel/data/config/Constants.java]: Right U R

**Pull request merged**